### ISSUE TITLE
DeepPavlov Bert 10x slower than Pytorch Pretrained Bert

### ISSUE THREAD

**Issue by lcswillems** [deepmipt/DeepPavlov#876]
Hi,

I modified the default DeepPavlov Bert to only output the pooled output (I followed the instructions given in [my previous issue](https://github.com/deepmipt/DeepPavlov/issues/825)).

However, this modified DeepPavlov Bert version is 10x slower than the [Pytorch Pretrained Bert](https://github.com/huggingface/pytorch-pretrained-BERT).

The code used for DeepPavlov Bert:

```
text = "Once when I was six years old I saw a magnificent picture in a book, called True Stories from Nature, about the primeval forest. It was a picture of a boa constrictor in the act of swallowing an animal. Here is a copy of the drawing."

model = deeppavlov.build_model('tools/deeppavlov/squad_bert.json')
embedding = model([text], [''])[0]
```

It takes 0.85 second to get the embedding.

Moreover, `model([text*10], [''])[0]` only takes 1 second, whereas `model([text]*10, ['']*10)[0]` takes 4 seconds. This should take the same time, isn't it?

**Comment by vikmary** [comment]
Hi,

The guess is that it is due to not optimal padding procedure. If you read [BertPreprocessor](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/models/preprocessors/bert_preprocessor.py#L76) in detail, you can see that every input sample is padded to `max_seq_length` of 512 tokens.

The maximum sequence length should be calculated dynamically for each batch individually. We believe to fix this in the next release.
